tests/cupy_tests/manipulation_tests/test_shape.py

```diff
@@ -170,6 +170,31 @@ def test_ravel3(self, xp, order):
         a = xp.asfortranarray(a)
         return a.ravel(order)
 
+    @testing.for_orders('CFA')
+    @testing.numpy_cupy_array_equal()
+    def test_ravel_non_contiguous(self, xp, order):
+        a = xp.arange(10)[::2]
+        assert not a.flags.c_contiguous and not a.flags.f_contiguous
+        b = a.ravel(order)
+        if order == 'C':
+            assert b.flags.c_contiguous
+        else:
+            assert b.flags.f_contiguous
```

Comment:
```suggestion
        assert b.flags.c_contiguous
```

The `order` argument is how to read the input.

`assert b.ndim == 1 and b.flags.c_contiguous` or `assert b.flags.c_contiguous and b.flags.f_contiguous` might state the intention clearer, while I think they are verbose.